proposals/0052-svg-components.md

```diff
@@ -0,0 +1,328 @@
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24" role="img">
+    <!-- SVG Content -->
+</svg>
+<svg width="20" height="24" role="img">
+    <!-- SVG Content -->
+</svg>
+<svg width="20" height="24" role="img">
+    <!-- SVG Content -->
+</svg>
+```
+
+## Accessibility Considerations
+
+One key aspect of this design is ensuring that the SVG components generated from `.svg` files are accessible by default, following best practices for web accessibility. This includes the following considerations:
+
+### ARIA Attributes
+
+- **Role Attribute:** By default, Astro could set `role="img"` on SVGs when they are used for non-decorative purposes.
+- **Title Element:** If the `.svg` file needs and accessible title/label, Astro can inject a `<title>` element.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<!-- Pass an accessible title that will be injected  -->
+<Logo title="Company Logo" />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24" role="img">
+    <title>Company Logo</title>
+    <!-- SVG Content -->
+</svg>
+```
+
+While this proposal offers strong defaults for accessibility, it is essential to give developers full control. Any automatically generated accessibility attributes (aria-label, role, etc.) should be overridable by the developer at the component level, ensuring that specific use cases or custom behaviors can be supported.
+
+# Testing Strategy
+
+In addition to standard unit and integration tests, the testing strategy should include:
+
+- **Cross-browser testing:** Ensuring that the Sprite pattern works consistently across all major browsers (e.g., Chrome, Firefox, Safari, Edge).
+- **Regression testing:** Verifying that this optimization doesn't unintentionally break any existing `.svg` imports.
+- **Performance testing:** Measuring the performance improvements in pages with many `<svg>` elements and comparing the impact of the new rendering system against inlining every SVG individually.
+
+# Drawbacks
+
+- **Increased Complexity for Debugging**: Debugging issues with SVG rendering could become more difficult due to the added abstraction of the Sprite approach. Developers may need additional documentation to understand how SVGs are being rendered and optimized, especially when diagnosing rendering problems in complex applications.
+- **Limited Use Cases:** The performance benefits of the Sprite approach may be minimal for projects that don't heavily rely on SVGs. For smaller projects or those with limited SVG usage, the optimization might introduce unnecessary complexity without providing significant gains in performance.
+- **Edge Cases with Advanced SVG Features:** Some advanced features of SVGs, like `<filter>` or `<mask>`, _may_ have issues when used within a `<symbol>` and referenced by `<use>`. These features will need thorough testing to ensure they function properly in the optimized model.
+
+# Alternatives
```

Comment:
Not discussed in the alternative is having a `<Svg />` component, similar to how we have an `<Image />` component rather than having the image import become a component. Going to bring this up with others on core we might remember better than me why we made that choice before.

Reply:
I've got prior art in https://github.com/jasikpark/astro-svg-loader

Reply:
Also consider to make it work with dynamic paths. Something like `./logos/${icon}.svg` is already working with the above suggested custom <SVG/> component. But more dynamic imports like `${BASE_DIR}${logo}.svg` fail to bundle correctly due to vite-rollup static analysis limitations.